File: JavaScript/Binary Tree - DFS/104. Maximum Depth of Binary Tree.js
/* 104. Maximum Depth of Binary Tree
Difficulty: Easy
Sumbission Link: https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/1386398870/
*/

var maxDepth = function (root)
 {
    if (!root) return 0

    let leftSubHeight = maxDepth(root.left)
    let rightSubHeight = maxDepth(root.right)
    
    return Math.max(leftSubHeight, rightSubHeight) + 1
};

File: Python/Binary Tree - DFS/104. Maximum Depth of Binary Tree.py
""" 104. Maximum Depth of Binary Tree
Difficulty: Easy
Sumbission Link: https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/1386399467/ """

class Solution(object):
    def maxDepth(self, root):
        if root == None:
            return 0
        return max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1